src/main.cpp

```diff
@@ -310,6 +312,16 @@ static void initTranslation()
         qDebug() << "Forcing locale" << locale << "from environment" << locale.uiLanguages();
     }
 
+    SettingsObject settings;
+    QString settingsLanguage( settings.read("ui.language").toString() );
+
+    if (!settingsLanguage.isEmpty()) {
+        locale = settingsLanguage;
+    } else {
+        //write an empty string to get "System default" language selected automatically in preferences
+        settings.write(QStringLiteral("ui.language"), QStringLiteral(""));
```

Comment:
use QString() instead of QStringLiteral("")

(do test this, but I think it should work)

QStringLiteral will allocate space in the compiled object for a string, which is really wasteful when there isn't actually any data to put in it :)